rpcs3/Emu/Cell/Modules/cellAtracXdec.cpp

```diff
@@ -0,0 +1,854 @@
+
+	return _CellAdecCoreOpOpenExt_atracx(ppu, handle, notifyAuDone, notifyAuDoneArg, notifyPcmOut, notifyPcmOutArg, notifyError, notifyErrorArg, notifySeqDone, notifySeqDoneArg, res, vm::null);
+}
+
+error_code _CellAdecCoreOpClose_atracx(ppu_thread& ppu, vm::ptr<AtracXdecContext> handle)
+{
+	std::unique_lock savestate_lock{g_fxo->get<hle_locks_t>(), std::try_to_lock};
+
+	if (!savestate_lock.owns_lock())
+	{
+		ppu.state += cpu_flag::again;
+		return {};
+	}
+
+	ppu.state += cpu_flag::wait;
+
+	cellAtracXdec.notice("_CellAdecCoreOpClose_atracx(handle=*0x%x)", handle);
+
+	ensure(!!handle); // Not checked on LLE
+
+	handle->run_thread = false;
+	handle->send_command<AtracXdecCmdType::close>(ppu);
+
+	{
+		std::lock_guard lock{handle->output_mutex};
+		handle->output_locked = false;
+	}
+
+	handle->output_consumed.notify_one();
+
+	if (vm::var<u64> ret; sys_ppu_thread_join(ppu, static_cast<u32>(handle->thread_id), +ret) != CELL_OK)
+	{
+		// Other thread already closed the decoder
+		return CELL_ADEC_ERROR_FATAL;
+	}
+
+	return CELL_OK;
+}
+
+error_code _CellAdecCoreOpStartSeq_atracx(ppu_thread& ppu, vm::ptr<AtracXdecContext> handle, vm::cptr<CellAdecParamAtracX> atracxParam)
+{
+	cellAtracXdec.notice("_CellAdecCoreOpStartSeq_atracx(handle=*0x%x, atracxParam=*0x%x)", handle, atracxParam);
+
+	ensure(!!handle && !!atracxParam); // Not checked on LLE
+
+	cellAtracXdec.notice("_CellAdecCoreOpStartSeq_atracx(): sampling_freq=%d, ch_config_idx=%d, nch_out=%d, nbytes=0x%x, extra_config_data=0x%08x, bw_pcm=0x%x, downmix_flag=%d, au_includes_ats_hdr_flg=%d",
+		atracxParam->sampling_freq, atracxParam->ch_config_idx, atracxParam->nch_out, atracxParam->nbytes, std::bit_cast<u32>(atracxParam->extra_config_data), atracxParam->bw_pcm, atracxParam->downmix_flag, atracxParam->au_includes_ats_hdr_flg);
+
+	return handle->send_command<AtracXdecCmdType::start_seq>(ppu, *atracxParam);
+}
+
+error_code _CellAdecCoreOpEndSeq_atracx(ppu_thread& ppu, vm::ptr<AtracXdecContext> handle)
+{
+	cellAtracXdec.notice("_CellAdecCoreOpEndSeq_atracx(handle=*0x%x)", handle);
+
+	ensure(!!handle); // Not checked on LLE
+
+	return handle->send_command<AtracXdecCmdType::end_seq>(ppu);
+}
+
+error_code _CellAdecCoreOpDecodeAu_atracx(ppu_thread& ppu, vm::ptr<AtracXdecContext> handle, s32 pcmHandle, vm::cptr<CellAdecAuInfo> auInfo)
+{
+	cellAtracXdec.trace("_CellAdecCoreOpDecodeAu_atracx(handle=*0x%x, pcmHandle=%d, auInfo=*0x%x)", handle, pcmHandle, auInfo);
+
+	ensure(!!handle && !!auInfo); // Not checked on LLE
+
+	cellAtracXdec.trace("_CellAdecCoreOpDecodeAu_atracx(): startAddr=*0x%x, size=0x%x, pts=%lld, userData=0x%llx", auInfo->startAddr, auInfo->size, std::bit_cast<be_t<u64>>(auInfo->pts), auInfo->userData);
+
+	return handle->send_command<AtracXdecCmdType::decode_au>(ppu, pcmHandle, *auInfo);
+}
+
+void _CellAdecCoreOpGetVersion_atracx(vm::ptr<std::array<u8, 4>> version)
```

Comment:
simply use vm::ptr<u8[4]>, I think its unspecifued behavior for std::array to match c array in byte size.